app/controllers/doorkeeper/authorizations_controller.rb

```diff
@@ -70,6 +70,7 @@ def pre_auth
 
     def pre_auth_params
       params.slice(*pre_auth_param_fields).permit(*pre_auth_param_fields)
+        .merge(current_resource_owner: current_resource_owner)
```

Comment:
I don't like this change just because we merge current user to params. Couldn't we move it in pre_authorization initializer somewhere?

Reply:
I did it this way so I didn't have to change the pre authorization initializer signature. I can make it an optional param to not affect other usages

Reply:
I don't really like multiple optional parameters though. Perhaps a named parameter? I'll let you decide

Reply:
I went with another optional param, the named param doesn't play nice with the `attrs` hash

Reply:
>  Perhaps a named parameter? I'll let you decide

For retro-compatibility we _must_ use optional parameter here. Maybe in some major version update I'll refactor this